sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,538 @@
+ feature-test macro always has this value.
+|===
+
+== Definitions
+
+== `task_sequence` class
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+
+template <auto &f, typename PropertyListT = empty_properties_t>
+class task_sequence {
+
+public:
+  task_sequence() = default;
+
+  task_sequence(const task_sequence &) = delete;
+  task_sequence &operator=(const task_sequence &) = delete;
+  task_sequence(task_sequence &&) = delete;
+  task_sequence &operator=(task_sequence &&) = delete;
+
+
+  void async(__unspecified__... args);
+
+  __unspecified__ get();
+
+  template<typename propertyT>
+  static constexpr bool has_property();
+
+  // The return type is an unspecified internal class used to represent
+  // instances of propertyT
+  template<typename propertyT>
+  static constexpr /*unspecified*/ get_property();
+
+  ~task_sequence();
+}
+
+} // namespace sycl::ext::intel::experimental
+
+----
+
+`task_sequence` is a class template, parameterized by an `auto` reference to a
+Callable `f`, and a list of properties _PropertyListT_. The Callable `f` defines
+the asynchronous task to be associated with the `task_sequence`, and requiring
+an auto reference ensures that each `f` is statically resolvable at compile
+time. Static resolvability by the compiler is desirable when compiling for
+spatial architectures as it can enable the generation of more efficient
+hardware. Templating a task_sequence on a non-Callable is malformed.
+_PropertyListT_ enables properties to be associated with a `task_sequence`
+
+[cols="1,1"]
+|===
+|Member functions of the `task_sequence` class|Description
+
+|void async(\\__unspecified__... args)
+|Available only in device functions. +
+ +
+ Asynchronously invoke the task_sequence's associated Callable `f` with `args`.
+ Calling `async` on a task_sequence to start a new invocation may block if the
+ number of outstanding invocations is equal to the invocation capacity.
+ Calling `async` on a task_sequence with mismatched argument types with respect
+ to the Callable `f`'s argument types is malformed.
+
+|\\__unspecified__ get()
+|Available only in device functions. +
+ +
+ Retrieves a return value from the task_sequence, which is the same return type
+ as the associated Callable `f`'s return type. Calling `get` on a task_sequence
+ to retrieve a return value from the task_sequence will block if there are no
+ return values to retrieve.
+|===
+
+By calling `async` on a `task_sequence` object more than once, the user implies
+that those invocations of `f` can be run in parallel. The implementation is,
+however, not obligated to run the invocations in parallel except in so far as is
+necessary to meet the forward progress guarantees outlined in the section on
+Progress Guarantees. It is undefined behaviour to template a task sequence
+object on a function and rely on communication between different invocations,
+i.e. through the use of atomics in the function.
```

Comment:
This is confusing.  First this part "to template a task sequence object on a function" is totally unnecessary as that's always how you use a task_sequence.  Secondly, why would it be UB to communicate between invocations?  So long as you use atomics and are cognizant of the forward progress guarantees it seems like a totally reasonable thing to do.